riscv/csrs.cc

```diff
@@ -465,24 +465,28 @@ bool sstatus_proxy_csr_t::unlogged_write(const reg_t val) noexcept {
 
 // implement class mstatus_csr_t
 mstatus_csr_t::mstatus_csr_t(processor_t* const proc, const reg_t addr):
-  base_status_csr_t(proc, addr),
-  val(0
-      | (proc->extension_enabled_const('U') && (proc->get_const_xlen() != 32) ? set_field((reg_t)0, MSTATUS_UXL, xlen_to_uxl(proc->get_const_xlen())) : 0)
-      | (proc->extension_enabled_const('S') && (proc->get_const_xlen() != 32) ? set_field((reg_t)0, MSTATUS_SXL, xlen_to_uxl(proc->get_const_xlen())) : 0)
-
+  base_status_csr_t(proc, addr) {
+#ifdef RISCV_ENABLE_DUAL_ENDIAN
+    const reg_t big_endian_bits = (proc->extension_enabled_const('U') ? MSTATUS_UBE : 0)
+                                | (proc->extension_enabled_const('S') ? MSTATUS_SBE : 0)
+                                | MSTATUS_MBE;
+#endif
+    val = 0
+          | (proc->extension_enabled_const('U') && (proc->get_const_xlen() != 32) ? set_field((reg_t)0, MSTATUS_UXL, xlen_to_uxl(proc->get_const_xlen())) : 0)
+          | (proc->extension_enabled_const('S') && (proc->get_const_xlen() != 32) ? set_field((reg_t)0, MSTATUS_SXL, xlen_to_uxl(proc->get_const_xlen())) : 0)
 #ifdef RISCV_ENABLE_DUAL_ENDIAN
-      | (proc->get_mmu()->is_target_big_endian() ? MSTATUS_UBE | MSTATUS_SBE | MSTATUS_MBE : 0)
+          | (proc->get_mmu()->is_target_big_endian() ? big_endian_bits : 0)
```

Comment:
Sorry, I did not realize when I requested this change that it was part of an initializer list. `val` really ought to be initialized in the initializer list, not in the constructor body.

You could use a [delegating constructor](https://stackoverflow.com/a/17375163/202201) though perhaps that's too much. Maybe there's a way to fit this all in one expression that would be easier to read than the original?

Reply:
Can we move all this into a separate function, and initialize the val with this function in the initializer list? just like the way for sstatus_write_mask of base_status_csr_t:
`sstatus_write_mask(compute_sstatus_write_mask())`

Reply:
Yes, good idea!